mycore-fo/src/main/java/org/mycore/component/fo/common/content/xml/MCRLayoutTransformerFoFactory.java

```diff
@@ -0,0 +1,160 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.component.fo.common.content.xml;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.TransformerException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.mycore.common.MCRClassTools;
+import org.mycore.common.MCRException;
+import org.mycore.common.config.MCRConfiguration;
+import org.mycore.common.content.transformer.MCRContentTransformer;
+import org.mycore.common.content.transformer.MCRContentTransformerFactory;
+import org.mycore.common.content.transformer.MCRIdentityTransformer;
+import org.mycore.common.content.transformer.MCRTransformerPipe;
+import org.mycore.common.content.transformer.MCRXSLTransformer;
+import org.mycore.common.xml.MCRLayoutTransformerFactory;
+import org.mycore.common.xml.MCRXMLResource;
+import org.mycore.component.fo.common.content.transformer.MCRFopper;
+import org.xml.sax.SAXException;
+
+import com.google.common.collect.Lists;
+import com.google.common.net.MediaType;
+
+/**
+ * This class acts as a {@link org.mycore.common.content.transformer.MCRContentTransformer} factory for 
+ * {@link org.mycore.common.xml.MCRLayoutService}.
+ * @author Thomas Scheffler (yagee)
+ *
+ */
+public class MCRLayoutTransformerFoFactory extends MCRLayoutTransformerFactory {
+    /** Map of transformer instances by ID */
+    private static HashMap<String, MCRContentTransformer> transformers = new HashMap<>();
+
+    private static Logger LOGGER = LogManager.getLogger(MCRLayoutTransformerFoFactory.class);
+
+    private static MCRFopper fopper = new MCRFopper();
+
+    private static final MCRIdentityTransformer NOOP_TRANSFORMER = new MCRIdentityTransformer("text/xml", "xml");
+
+    /**
+     * Returns the transformer with the given ID. If the transformer is not instantiated yet,
+     * it is created and initialized.
+     */
+    @Override
+    public MCRContentTransformer getTransformer(String id) throws Exception {
+        MCRContentTransformer transformer = transformers.get(id);
+        if (transformer != null) {
+            return transformer;
+        }
+        //try to get configured transformer
+        transformer = MCRContentTransformerFactory.getTransformer(id.replaceAll("-default$", ""));
+        if (transformer != null) {
+            transformers.put(id, transformer);
+            return transformer;
+        }
+        return buildLayoutTransformer(id);
+    }
+
+    private static MCRContentTransformer buildLayoutTransformer(String id) throws Exception {
```

Comment:
I dont think this method is needed.(soo commend above)

Reply:
This method is different to against the super class and supported the PDF media type.

Reply:
Yes i know but the difference is to small to copy the entire method

Reply:
![image](https://user-images.githubusercontent.com/7668803/61540153-6114f080-aa3d-11e9-92aa-609931450b7a.png)
This is the entire difference between both classes.